nusantara/nusa_datasets/id_hsd_nofaaulia/id_hsd_nofaaulia.py

```diff
@@ -0,0 +1,178 @@
+"""
+
+_HOMEPAGE = "https://github.com/nofaulia/hate-speech-detection"
+
+_LICENSE = "Unknown"
+
+_URLS = {
+    _DATASETNAME: "https://raw.githubusercontent.com/nofaulia/hate-speech-detection/main/data/dataset.csv",
+}
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]  # example: [Tasks.TRANSLATION, Tasks.NAMED_ENTITY_RECOGNITION, Tasks.RELATION_EXTRACTION]
+
+_SOURCE_VERSION = "1.0.0"
+_NUSANTARA_VERSION = "1.0.0"
+
+
+# TODO: Name the dataset class to match the script name using CamelCase instead of snake_case
+class NewDataset(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    NUSANTARA_VERSION = datasets.Version(_NUSANTARA_VERSION)
+
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="id_hsd_nofaaulia_source",
+            version=SOURCE_VERSION,
+            description="id_hsd_nofaaulia source schema",
+            schema="source",
+            subset_id="id_hsd_nofaaulia",
+        ),
+        NusantaraConfig(
+            name="id_hsd_nofaaulia_nusantara_text",
+            version=NUSANTARA_VERSION,
+            description="id_hsd_nofaaulia Nusantara schema",
+            schema="nusantara_text",
+            subset_id="id_hsd_nofaaulia",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "id_hsd_nofaaulia_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features({"index": datasets.Value("string"), "text": datasets.Value("string"), "label": datasets.Value("string")})
+        elif self.config.schema == "nusantara_text":
+            features = schemas.text_features(["0", "1"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = _URLS[_DATASETNAME]
+        data_dir = Path(dl_manager.download_and_extract(urls))
+
+        data_files = {"train": os.path.join(data_dir.parent, "train.csv"), "test": os.path.join(data_dir.parent, "test.csv")}
+
+        df = pd.read_csv(data_dir)
+        df = df.dropna(axis=0)
+
+        features = [c for c in df.columns.values if c not in []]
+        target = "label"
+
+        # The split follows the implementation below
+        # https://github.com/IndoNLP/nusa-crowd/blob/master/nusantara/utils/schemas/pairs.py
+        # test_size=0.1, random_state=42
+        x_train, x_test, y_train, y_test = train_test_split(df[features], df[target], test_size=0.1, random_state=42)
```

Comment:
I think if the dataset doesn't have a predefined canonical train/test split, we just need to put all data into split.TRAIN

Reply:
Umm, I am confused. According to [the note](https://indonlp.github.io/nusa-catalogue/card.html?id_hsd_nofaaulia), the split should yield 816 train, 0 validation, and 91 test. Moreover, the implementation done following the splitting mechanism mentioned [here](https://github.com/IndoNLP/nusa-crowd/issues/225) by @SamuelCahyawijaya. 

Which one should I follow?

Reply:
Ah didn't notice there's a comment there. Yea I think we can go with this config. Even though I still think we need to make sure that the split is consistent (from my previous experience different versions of library, python or OS might result in different randomization even though we've set the same random state). Curious about what other reviewers think

Reply:
Hi @christianwbsn, I never encountered the problem before actually, and I do think if someone uses our dataloader to reproduce original results and didn't find it the same, he/she can also check the splitting on the original source link.

Could @IvanHalimP maybe also help to pin down the version of sklearn that you used in the `requirements.txt` (since it's needed anyways) and confirm that all ids in the splits are the same? and could you also mention your version of python that you use to confirm too?

If that's done, I do think it's quite robust enough.
Thank you @IvanHalimP and @christianwbsn !

Reply:
The current libraries' versions are as follows:
scikit-learn==1.1.2
python==3.10.4

Checked the transcripts by running it multiple times (I did it 6 times) with the same random state, it returned the same transcripts. Tried to change the random state and it was still consistent. I don't know if the test I performed is valid though.

Reply:
Hello all, sorry I'm late to the party. :partying_face: I think the datasheet in NusaCatalogue should be 815 train + 91 test + 1 header (not counted) because there are 907 rows in the original dataset. The datasheet submitter also put the total data as 906 (not counting the header). I guess they just made a mistake during listing the train-test split.

I have updated the datasheet in NusaCatalogue so it will be consistent with this dataloader's result.

Reply:
I think your test is valid @IvanHalimP, thank you for providing the versions of the library too for reproducability!

Thank you for contributing, @IvanHalimP !